segpy/binary_reel_header.py

```diff
@@ -1,5 +1,73 @@
 from segpy.header import FormatMeta, field
-from segpy.field_types import Int32, Int16
+from segpy.field_types import IntFieldMeta, IntEnumFieldMeta, Int32, Int16, UInt32, UInt16
+
+
+class DataSampleFormatField(metaclass=IntEnumFieldMeta,
+                            values=(range(9))):
+    pass
+
+
+class TraceSortingField(metaclass=IntEnumFieldMeta,
+                        values=range(-1, 10)):
+    pass
+
+
+class SweepTypeField(metaclass=IntEnumFieldMeta,
+                     values=range(5)):
+    pass
+
+
+class TaperTypeField(metaclass=IntEnumFieldMeta,
+                     values=range(4)):
+    pass
+
+
+class CorrelatedDataTracesField(metaclass=IntEnumFieldMeta,
+                                values=range(3)):
+    pass
+
+
+class BinaryGainRecoveredField(metaclass=IntEnumFieldMeta,
+                               values=range(3)):
+    pass
+
+
+class AmplitudeRecoveryMethodField(metaclass=IntEnumFieldMeta,
+                                   values=range(5)):
+    pass
+
+
+class MeasurementSystemField(metaclass=IntEnumFieldMeta,
+                             values=range(3)):
+    pass
+
+
+class ImpulseSignalPolarityField(metaclass=IntEnumFieldMeta,
+                                 values=range(3)):
+    pass
+
+
+class VibratoryPolarityCodeField(metaclass=IntEnumFieldMeta,
+                                 values=range(9)):
+    pass
+
+
+class FixedLengthTraceFlag(metaclass=IntEnumFieldMeta,
+                           values=range(2)):
+    pass
+
+
+class NumExtendedTextualHeadersField(metaclass=IntFieldMeta,
+                                     seg_y_type='int16',
+                                     min_value=-1):
+    pass
+
+
+class FormatRevisionNumField(metaclass=IntEnumFieldMeta,
+                             seg_y_type='int16',
+                             values=(0x0100, 0x0000)):
```

Comment:
We need to be carefull here, just because this field is so commonly screwed up.  You can see some attempt to deal with the dysfunction seen in the field in `segpy.revisions.py`.  At the very least the constants 0x0100 and 0x0000 are defined there and should be reused here.

Reply:
I think it's right that we be strict in interpreting these at this level though.